pkg/repository/v1/task.go

```diff
@@ -2422,13 +2423,25 @@ func makeEventTypeArr(status sqlcv1.V1TaskEventType, n int) []sqlcv1.V1TaskEvent
 	return a
 }
 
+func hash(s string) int64 {
+	h := fnv.New64a()
+	h.Write([]byte(s))
+	return int64(h.Sum64())
+}
+
 func (r *TaskRepositoryImpl) ReplayTasks(ctx context.Context, tenantId string, tasks []TaskIdInsertedAtRetryCount) (*ReplayTasksResult, error) {
 	tx, commit, rollback, err := sqlchelpers.PrepareTx(ctx, r.pool, r.l, 30000)
 
 	if err != nil {
 		return nil, err
 	}
 
+	err = r.queries.AdvisoryLock(ctx, tx, hash(tenantId))
```

Comment:
Have we confirmed that we don't have a collision with how we're computing some other advisory lock (like in the concurrency logic)? Might make sense to prefix `replay_` to the tenant ID here or smth

Reply:
good call, will fix that